plugins/common/tls/utils.go

```diff
@@ -36,3 +43,58 @@ func ParseTLSVersion(version string) (uint16, error) {
 	sort.Strings(available)
 	return 0, fmt.Errorf("unsupported version %q (available: %s)", version, strings.Join(available, ","))
 }
+
+func readFile(filename string) []byte {
+	octets, err := os.ReadFile(filename)
+	if err != nil {
+		panic(fmt.Sprintf("reading %q: %v", filename, err))
+	}
+	return octets
+}
+
+func ReadCertificate(filename string) string {
+	octets := readFile(filename)
+	return string(octets)
+}
+
+func ReadKey(filename string, password string) string {
+	keyBytes := readFile(filename)
+	currentBlock, remainingBlocks := pem.Decode(keyBytes)
+	if currentBlock == nil {
+		panic(errors.New("failed to decode private key: no PEM data found"))
+	}
+	var allBlocks string
+	for {
+		if currentBlock.Type == "ENCRYPTED PRIVATE KEY" {
+			if password == "" {
+				panic(errors.New("missing password for PKCS#8 encrypted private key"))
+			}
+			var decryptedKey *rsa.PrivateKey
+			decryptedKey, err := pkcs8.ParsePKCS8PrivateKeyRSA(currentBlock.Bytes, []byte(password))
+			if err != nil {
+				panic(fmt.Errorf("failed to parse encrypted PKCS#8 private key: %w", err))
+			}
+			pemBlock := string(pem.EncodeToMemory(&pem.Block{Type: currentBlock.Type, Bytes: x509.MarshalPKCS1PrivateKey(decryptedKey)}))
+			allBlocks += pemBlock
+		} else if currentBlock.Headers["Proc-Type"] == "4,ENCRYPTED" {
+			decryptedKeyDER, err := x509.DecryptPEMBlock(currentBlock, []byte(password))
+			if err != nil {
+				panic(fmt.Errorf("failed to parse encrypted private key %w", err))
+			}
+			decryptedKey, err := x509.ParsePKCS1PrivateKey(decryptedKeyDER)
+			if err != nil {
+				panic(fmt.Errorf("unable to convert from DER to PEM format: %w", err))
+			}
+			pemBlock := string(pem.EncodeToMemory(&pem.Block{Type: currentBlock.Type, Bytes: x509.MarshalPKCS1PrivateKey(decryptedKey)}))
+			allBlocks += pemBlock
+		} else {
+			pemBlock := string(pem.EncodeToMemory(&pem.Block{Type: currentBlock.Type, Bytes: currentBlock.Bytes}))
+			allBlocks += pemBlock
+		}
+		currentBlock, remainingBlocks = pem.Decode(remainingBlocks)
+		if currentBlock == nil {
+			break
+		}
+	}
+	return allBlocks
```

Comment:
IIUC, `allblocks` will return concatenated strings of multiple blocks, which I think is wrong since a) with a combined file, `allblocks` will have both the private key and the cert in it, which would be unexpected with the second argument of `tls.X509KeyPair()` and b) `allblocks` will contain multiple private keys if the key file has them, which is also unexpected with `tls.X509KeyPair()`. https://pkg.go.dev/crypto/tls#X509KeyPair suggests that the second argument should only be a single private block. I suggest not relying on implementation dependent behavior in `tls.X509KeyPair()` and make this code only return one private block.

Since this is called `ReadKey()`, I think it makes sense to:
* ignore non-private keys instead of appending them
* return the first private key rather than appending them (with a code comment)